src/main/java/com/example/may/cloudfoundry/destination/service/DestinationService.java

```diff
@@ -0,0 +1,96 @@
+package com.example.may.cloudfoundry.destination.service;
+
+import com.example.may.email.config.model.EmailProperties;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.json.JSONObject;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.retry.annotation.Backoff;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException.Unauthorized;
+import org.springframework.web.client.RestTemplate;
+
+import static com.example.may.core.ouath.constant.OAuth2Constants.BEARER_AUTHORIZATION_TYPE;
+import static com.example.may.core.ouath.constant.OAuth2Constants.DESTINATION_CONFIGURATION_RESPONSE_FIELD;
+import static com.example.may.core.ouath.constant.OAuth2Constants.HEADER_NAME_AUTHORIZATION;
+import static com.example.may.email.config.model.EmailConstant.MAIL_FROM_RESPONSE_FIELD;
+import static com.example.may.email.config.model.EmailConstant.MAIL_HOST_RESPONSE_FIELD;
+import static com.example.may.email.config.model.EmailConstant.MAIL_PASSWORD_RESPONSE_FIELD;
+import static com.example.may.email.config.model.EmailConstant.MAIL_PORT_RESPONSE_FIELD;
+import static com.example.may.email.config.model.EmailConstant.MAIL_USER_RESPONSE_FIELD;
+import static java.lang.String.format;
+
+/**
+ * @author Evelina Tubalets
+ */
+@Slf4j
+@Service
+@AllArgsConstructor
+public class DestinationService {
+
+    public static final String MAIL_DESTINATION_PATH = "/destination-configuration/v1/destinations/mail-destination";
+
+    private final DestinationXsuaaTokenService destinationXsuaaTokenService;
+    private final RestTemplate restTemplate;
+
+    @Retryable(value = Unauthorized.class, maxAttempts = 2, backoff = @Backoff(0))
+    public EmailProperties getEmailProperties() {
```

Comment:
It should return generic type, which will contain properties from destination. And this service should be reusable by different app components